operator/pkg/controller/virtualip/controller.go

```diff
@@ -293,3 +309,80 @@ func (r *VirtualIPReconciler) mutateConfigMap(obj *corev1.ConfigMap) error {
 
 	return nil
 }
+
+func (r *VirtualIPReconciler) mutateDaemonSet(obj *appsv1.DaemonSet) error {
+	poolName := obj.GetLabels()[labelPoolName]
+	poolInfo := r.getPools()[poolName]
+
+	selector := make(client.MatchingLabels)
+	selector[labelPoolName] = poolName
+
+	obj.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: selector,
+	}
+	utils.UpdateLabels(&obj.Spec.Template.ObjectMeta, selector)
+
+	obj.Spec.Template.Spec.NodeSelector = poolInfo.NodeSelector
+
+	obj.Spec.Template.Spec.HostNetwork = true
+
+	volumeName := "hl-config"
+	defaultMode := int32(420)
+	obj.Spec.Template.Spec.Volumes = []corev1.Volume{
+		{
+			Name: volumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: r.getObjName(poolName),
+					},
+					DefaultMode: &defaultMode,
+				},
+			},
+		},
+	}
+
+	// If we have more than 1 container we clean up the containers list
```

Comment:
Guess that's fine for now, but why do we need to do this? Can't we just select by name? I'm thinking we might want to let admins add some sidecar, but not sure... Anyway, that's nitpicking, no need to change anything until the need comes up.

Reply:
We could use the name to retrieve the right container, but, to me, we manage this DaemonSet so the user is not supposed to change it, if we want to support sideCars we will want to expose it in our ClusterConfig